test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs

```diff
@@ -78,5 +75,54 @@ public void DotnetRestore_SolutionRestoreVerifySolutionDirPassedToProjects()
                 result.AllOutput.Should().Contain($"|SOLUTION {PathUtility.EnsureTrailingSlash(pathContext.SolutionRoot)} proj|");
             }
         }
+
+        [PlatformFact(Platform.Windows)]
+        public void DotnetRestore_WithAuthorSignedPackage_Succeeds()
+        {
+            using (var packageSourceDirectory = TestDirectory.Create())
+            using (var testDirectory = TestDirectory.Create())
+            {
+                var packageFile = new FileInfo(Path.Combine(packageSourceDirectory.Path, "TestPackage.AuthorSigned.1.0.0.nupkg"));
+                var package = GetResource(packageFile.Name);
+
+                File.WriteAllBytes(packageFile.FullName, package);
+
+                var projectName = "ClassLibrary1";
+                var workingDirectory = Path.Combine(testDirectory, projectName);
+                var projectFile = Path.Combine(workingDirectory, $"{projectName}.csproj");
+
+                _msbuildFixture.CreateDotnetNewProject(testDirectory.Path, projectName, " classlib");
+
+                using (var stream = File.Open(projectFile, FileMode.Open, FileAccess.ReadWrite))
+                {
+                    var xml = XDocument.Load(stream);
+
+                    ProjectFileUtils.SetTargetFrameworkForProject(xml, "TargetFrameworks", "netstandard1.3");
+
+                    var attributes = new Dictionary<string, string>() { { "Version", "1.0.0" } };
+
+                    ProjectFileUtils.AddItem(
+                        xml,
+                        "PackageReference",
+                        "TestPackage.AuthorSigned",
+                        "netstandard1.3",
+                        new Dictionary<string, string>(),
+                        attributes);
+
+                    ProjectFileUtils.WriteXmlToFile(xml, stream);
+                }
+
+                var args = $"--source \"{packageSourceDirectory.Path}\" ";
+
+                _msbuildFixture.RestoreProject(workingDirectory, projectName, args);
```

Comment:
this test is not asserting anything.

Reply:
The MsbuildIntegrationTestFixture asserts the exit code inside, doesn't it? :)
You should know 😃

Reply:
As @nkolev92 said, `MsbuildIntegrationTestFixture .RestoreProject(...)` [asserts](https://github.com/NuGet/NuGet.Client/blob/023fe7670796a8986bbfdc520029e4cf0a6bbfda/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuilldIntegrationTestFixture.cs#L143) restore success.